src/runtime-rs/crates/hypervisor/src/device/util.rs

```diff
@@ -69,9 +69,55 @@ pub(crate) fn get_virt_drive_name(mut index: i32) -> Result<String> {
     Ok(String::from(PREFIX) + std::str::from_utf8(&disk_letters)?)
 }
 
+// do_increase_count
+pub fn do_increase_count(ref_count: &mut u64) -> Result<bool> {
+    match *ref_count {
+        0 => {
+            // When ref_count is 0, it indicates that the device is new here, In this case, we know that the device
+            // has not been attached into the Guest.
+            *ref_count += 1;
+            Ok(false)
+        }
+        // While in practice, the number of times a device is attempted to be inserted into the Guest cannot reach
+        // this maximum value, the device is theoretically allowed to do so.
+        // However, we will not allow the device to actually be inserted into the Guest more than once.
+        std::u64::MAX => Err(anyhow!("device was attached too many times")),
+        _ => {
+            // When ref_count is greater than 0, it indicates that how many times the device has been attempted to
+            // be inserted into the Guest.
+            // In this case, we know the device has been attached into the Guest.
+            *ref_count += 1;
+            Ok(true)
+        }
```

Comment:
How about using `checked_add()`? FYI, https://doc.rust-lang.org/std/primitive.u32.html#method.checked_add.

Reply:
Thx @justxuewei Although I know that the `checked_add()` method is a good choice in some cases, I don't think it's appropriate here. And I don't want to introduce too much judgment logic because of this method, so I choose to stick to the original approach.

Another reason is, the code are just moved here from its original places, and the key point is, simple and it works well.

Reply:
@justxuewei  Maybe saturating_add(1)/saturating_sub(1) is a good choice !

Reply:
However, It's indeed better to refactor it with check_add as @justxuewei said:
```
checked_add(1).ok_or(anyhow!("error msg")?;
return Ok(*ref_count != 1))
```